cmd/snap/cmd_debug_api.go

```diff
@@ -0,0 +1,108 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/url"
+	"strings"
+
+	"github.com/jessevdk/go-flags"
+	"github.com/snapcore/snapd/logger"
+)
+
+type cmdDebugAPI struct {
+	clientMixin
+
+	Pretty bool `long:"pretty"`
+
+	Positional struct {
+		Method  string   `positional-arg-name:"<method>"`
+		Query   string   `positional-arg-name:"<query>"`
```

Comment:
this is a bit of a strange as it contains also the path?

Reply:
Renamed the argument. I didn't want to split it into path and query params, such that `snap debug api GET /v2/snaps` works and you need to pass additional query args it's simply `snap debug api GET '/v2/find?name=foo'`

Reply:
I've tweaked the arguments now so that we're closer to what curl expects, i.e.:

```
Usage:
  test-snap debug api [api-OPTIONS] <path-and-query>

...
[api command options]
          --pretty            Pretty print if response is JSON
      -H, --header=           Set header (can be repeated multiple times)
      -X, --request=          HTTP method to use (defaults to GET)
          --fail              Fail on request errors
```